src/vm/gdbjit.cpp

```diff
@@ -106,13 +107,15 @@ GetTypeInfoFromTypeHandle(TypeHandle typeHandle, NotifyGdb::PTK_TypeInfoMap pTyp
                 else
                 {
                     PTR_BYTE base = 0;
-                    if (!pField->IsRVA())
+                    MethodTable* pMT = pField->GetEnclosingMethodTable();
+                    base = pField->GetBaseInDomainLocalModule(pMT->GetDomainLocalModule(NULL));
```

Comment:
Ah, I see.
Hmm, it is strange that it works for DACCESS_COMPILE whern you pass in NULL AppDomain parameter though. Looking at the MethodTable::GetDomainLocalModule(AppDomain * pAppDomain), I can see it calls GetModuleForStatics()->GetDomainLocalModule(pAppDomain); and the Module::GetDomainLocalModule(AppDomain *pDomain) asserts when DACCESS_COMPILE is defined and the AppDomain pointer is NULL (unless the first "if" in the function causes it to return before doing anything with the AppDomain pointer.